packages/library-base/src/commonMain/kotlin/io/realm/DynamicRealmObject.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import kotlin.reflect.KClass
+
+/**
+ * A **dynamic realm object** gives access the data of the realm objects through a generic string
+ * based API instead of the conventional [Realm] API that only allows access through the properties
+ * of the corresponding schema classes supplied in the configuration.
+ */
+interface DynamicRealmObject : RealmObject {
+    /**
+     * The name of the class in the realm object model.
```

Comment:
Suggested rewrite (taken from Java):

```
     * The type of the object. This will normally correspond to the name of a model class that is extending
     * [io.realm.RealmObject].
```